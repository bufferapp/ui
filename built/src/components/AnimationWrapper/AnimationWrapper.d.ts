import React from 'react';
import PropTypes from 'prop-types';
declare const AnimationWrapper: {
    ({ align, children, dismissing, duration, easing, justify, stageInAnimation, stageOutAnimation, onDismiss, }: {
        align: any;
        children: any;
        dismissing: any;
        duration: any;
        easing: any;
        justify: any;
        stageInAnimation: any;
        stageOutAnimation: any;
        onDismiss: any;
    }): React.JSX.Element;
    propTypes: {
        align: PropTypes.Requireable<string>;
        children: PropTypes.Validator<string | number | boolean | PropTypes.ReactElementLike | PropTypes.ReactNodeArray>;
        dismissing: PropTypes.Requireable<boolean>;
        duration: PropTypes.Requireable<number>;
        justify: PropTypes.Requireable<string>;
        easing: PropTypes.Requireable<string>;
        stageInAnimation: PropTypes.Validator<PropTypes.InferProps<{}>>;
        stageOutAnimation: PropTypes.Validator<PropTypes.InferProps<{}>>;
        onDismiss: PropTypes.Requireable<(...args: any[]) => any>;
    };
    defaultProps: {
        align: string;
        dismissing: boolean;
        duration: number;
        justify: string;
        easing: string;
        onDismiss: () => void;
    };
};
export default AnimationWrapper;
//# sourceMappingURL=AnimationWrapper.d.ts.map