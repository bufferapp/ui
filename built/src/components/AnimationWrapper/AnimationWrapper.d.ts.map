{"version":3,"file":"AnimationWrapper.d.ts","sourceRoot":"","sources":["../../../../src/components/AnimationWrapper/AnimationWrapper.tsx"],"names":[],"mappings":"AAAA,OAAO,KAA8B,MAAM,OAAO,CAAA;AAClD,OAAO,SAAS,MAAM,YAAY,CAAA;AA8BlC,QAAA,MAAM,gBAAgB;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;CAqFrB,CAAA;AAuBD,eAAe,gBAAgB,CAAA"}