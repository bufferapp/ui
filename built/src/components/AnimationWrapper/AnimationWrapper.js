import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { easeOutQuart } from '../style/animations';
const AnimationContainer = styled.div `
  animation-fill-mode: forwards;
  display: flex;
  align-items: ${({ align }) => align}}
  justify-content: ${({ justify }) => justify}}
  outline: none;
  width: 100%;

  &.stageIn {
    animation: ${({ duration }) => `${duration}ms`} ${({ stageInAnimation }) => stageInAnimation} ${({ easing }) => easing};
  }

  &.stageOut {
    animation: ${({ duration }) => `${duration}ms`} ${({ stageOutAnimation }) => stageOutAnimation} ${({ easing }) => easing};
  }

  @media (prefers-reduced-motion) {
    &.stageIn,
    &.stageOut {
      animation-name: dissolve;
    }
  }
`;
const AnimationWrapper = ({ align, children, dismissing, duration, easing, justify, stageInAnimation, stageOutAnimation, onDismiss, }) => {
    const [content, setContent] = useState(children);
    const [hasChanged, setHasChanged] = useState(false);
    const [dismissed, setDismissed] = useState(false);
    const [animationLocked, setAnimationLocked] = useState(false);
    const [className, setClassName] = useState('stageIn');
    const shouldUpdate = children !== content && !dismissing;
    function lockAnimationDuringStaging() {
        setAnimationLocked(true);
        setTimeout(() => {
            setAnimationLocked(false);
        }, duration * 2);
    }
    function stageIn() {
        setClassName('stageIn');
    }
    function stageOut() {
        setClassName('stageOut');
    }
    useEffect(() => {
        if (shouldUpdate) {
            setHasChanged(true);
            setTimeout(() => {
                stageIn();
                setContent(children);
                setHasChanged(false);
            }, duration);
        }
    }, [children]);
    useEffect(() => {
        if (dismissing) {
            setTimeout(() => {
                setDismissed(true);
                onDismiss();
            }, duration - 10);
        }
    }, [dismissing]);
    useEffect(() => {
        lockAnimationDuringStaging();
    }, []);
    if (!animationLocked && (hasChanged || dismissing)) {
        stageOut();
        lockAnimationDuringStaging();
    }
    return (React.createElement(AnimationContainer, { align: align, duration: duration, easing: easing, justify: justify, stageInAnimation: stageInAnimation, stageOutAnimation: stageOutAnimation, className: className }, !dismissed && content));
};
AnimationWrapper.propTypes = {
    align: PropTypes.string,
    children: PropTypes.node.isRequired,
    dismissing: PropTypes.bool,
    duration: PropTypes.number,
    justify: PropTypes.string,
    easing: PropTypes.string,
    stageInAnimation: PropTypes.shape({}).isRequired,
    stageOutAnimation: PropTypes.shape({}).isRequired,
    onDismiss: PropTypes.func,
};
AnimationWrapper.defaultProps = {
    align: 'center',
    dismissing: false,
    duration: 300,
    justify: 'center',
    easing: easeOutQuart,
    onDismiss: () => { },
};
export default AnimationWrapper;
//# sourceMappingURL=AnimationWrapper.js.map