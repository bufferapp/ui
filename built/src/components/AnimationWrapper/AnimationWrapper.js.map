{"version":3,"file":"AnimationWrapper.js","sourceRoot":"","sources":["../../../../src/components/AnimationWrapper/AnimationWrapper.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,SAAS,EAAE,QAAQ,EAAE,MAAM,OAAO,CAAA;AAClD,OAAO,SAAS,MAAM,YAAY,CAAA;AAClC,OAAO,MAAM,MAAM,mBAAmB,CAAA;AACtC,OAAO,EAAE,YAAY,EAAE,MAAM,qBAAqB,CAAA;AAElD,MAAM,kBAAkB,GAAG,MAAM,CAAC,GAAG,CAAA;;;iBAGpB,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,CAAC,KAAK;qBAChB,CAAC,EAAE,OAAO,EAAE,EAAE,EAAE,CAAC,OAAO;;;;;iBAK5B,CAAC,EAAE,QAAQ,EAAE,EAAE,EAAE,CAAC,GAAG,QAAQ,IAAI,IAAI,CAAC,EAAE,gBAAgB,EAAE,EAAE,EAAE,CAC7E,gBAAgB,IAAI,CAAC,EAAE,MAAM,EAAE,EAAE,EAAE,CAAC,MAAM;;;;iBAI3B,CAAC,EAAE,QAAQ,EAAE,EAAE,EAAE,CAAC,GAAG,QAAQ,IAAI,IAAI,CAAC,EAAE,iBAAiB,EAAE,EAAE,EAAE,CAC9E,iBAAiB,IAAI,CAAC,EAAE,MAAM,EAAE,EAAE,EAAE,CAAC,MAAM;;;;;;;;;CAS5C,CAAA;AAED,MAAM,gBAAgB,GAAG,CAAC,EAExB,KAAK,EAEL,QAAQ,EAER,UAAU,EAEV,QAAQ,EAER,MAAM,EAEN,OAAO,EAEP,gBAAgB,EAEhB,iBAAiB,EAEjB,SAAS,GACV,EAAE,EAAE;IACH,MAAM,CAAC,OAAO,EAAE,UAAU,CAAC,GAAG,QAAQ,CAAC,QAAQ,CAAC,CAAA;IAChD,MAAM,CAAC,UAAU,EAAE,aAAa,CAAC,GAAG,QAAQ,CAAC,KAAK,CAAC,CAAA;IACnD,MAAM,CAAC,SAAS,EAAE,YAAY,CAAC,GAAG,QAAQ,CAAC,KAAK,CAAC,CAAA;IACjD,MAAM,CAAC,eAAe,EAAE,kBAAkB,CAAC,GAAG,QAAQ,CAAC,KAAK,CAAC,CAAA;IAC7D,MAAM,CAAC,SAAS,EAAE,YAAY,CAAC,GAAG,QAAQ,CAAC,SAAS,CAAC,CAAA;IACrD,MAAM,YAAY,GAAG,QAAQ,KAAK,OAAO,IAAI,CAAC,UAAU,CAAA;IAExD,SAAS,0BAA0B;QACjC,kBAAkB,CAAC,IAAI,CAAC,CAAA;QACxB,UAAU,CAAC,GAAG,EAAE;YACd,kBAAkB,CAAC,KAAK,CAAC,CAAA;QAC3B,CAAC,EAAE,QAAQ,GAAG,CAAC,CAAC,CAAA;IAClB,CAAC;IAED,SAAS,OAAO;QACd,YAAY,CAAC,SAAS,CAAC,CAAA;IACzB,CAAC;IAED,SAAS,QAAQ;QACf,YAAY,CAAC,UAAU,CAAC,CAAA;IAC1B,CAAC;IAED,SAAS,CAAC,GAAG,EAAE;QACb,IAAI,YAAY,EAAE;YAChB,aAAa,CAAC,IAAI,CAAC,CAAA;YACnB,UAAU,CAAC,GAAG,EAAE;gBACd,OAAO,EAAE,CAAA;gBACT,UAAU,CAAC,QAAQ,CAAC,CAAA;gBACpB,aAAa,CAAC,KAAK,CAAC,CAAA;YACtB,CAAC,EAAE,QAAQ,CAAC,CAAA;SACb;IACH,CAAC,EAAE,CAAC,QAAQ,CAAC,CAAC,CAAA;IAEd,SAAS,CAAC,GAAG,EAAE;QACb,IAAI,UAAU,EAAE;YAEd,UAAU,CAAC,GAAG,EAAE;gBACd,YAAY,CAAC,IAAI,CAAC,CAAA;gBAClB,SAAS,EAAE,CAAA;YACb,CAAC,EAAE,QAAQ,GAAG,EAAE,CAAC,CAAA;SAClB;IACH,CAAC,EAAE,CAAC,UAAU,CAAC,CAAC,CAAA;IAEhB,SAAS,CAAC,GAAG,EAAE;QACb,0BAA0B,EAAE,CAAA;IAC9B,CAAC,EAAE,EAAE,CAAC,CAAA;IAEN,IAAI,CAAC,eAAe,IAAI,CAAC,UAAU,IAAI,UAAU,CAAC,EAAE;QAClD,QAAQ,EAAE,CAAA;QACV,0BAA0B,EAAE,CAAA;KAC7B;IAED,OAAO,CACL,oBAAC,kBAAkB,IACjB,KAAK,EAAE,KAAK,EACZ,QAAQ,EAAE,QAAQ,EAClB,MAAM,EAAE,MAAM,EACd,OAAO,EAAE,OAAO,EAChB,gBAAgB,EAAE,gBAAgB,EAClC,iBAAiB,EAAE,iBAAiB,EACpC,SAAS,EAAE,SAAS,IAEnB,CAAC,SAAS,IAAI,OAAO,CACH,CACtB,CAAA;AACH,CAAC,CAAA;AAED,gBAAgB,CAAC,SAAS,GAAG;IAC3B,KAAK,EAAE,SAAS,CAAC,MAAM;IACvB,QAAQ,EAAE,SAAS,CAAC,IAAI,CAAC,UAAU;IACnC,UAAU,EAAE,SAAS,CAAC,IAAI;IAC1B,QAAQ,EAAE,SAAS,CAAC,MAAM;IAC1B,OAAO,EAAE,SAAS,CAAC,MAAM;IACzB,MAAM,EAAE,SAAS,CAAC,MAAM;IACxB,gBAAgB,EAAE,SAAS,CAAC,KAAK,CAAC,EAAE,CAAC,CAAC,UAAU;IAChD,iBAAiB,EAAE,SAAS,CAAC,KAAK,CAAC,EAAE,CAAC,CAAC,UAAU;IACjD,SAAS,EAAE,SAAS,CAAC,IAAI;CAC1B,CAAA;AAED,gBAAgB,CAAC,YAAY,GAAG;IAC9B,KAAK,EAAE,QAAQ;IACf,UAAU,EAAE,KAAK;IACjB,QAAQ,EAAE,GAAG;IACb,OAAO,EAAE,QAAQ;IACjB,MAAM,EAAE,YAAY;IACpB,SAAS,EAAE,GAAG,EAAE,GAAE,CAAC;CACpB,CAAA;AAED,eAAe,gBAAgB,CAAA","sourcesContent":["import React, { useEffect, useState } from 'react'\nimport PropTypes from 'prop-types'\nimport styled from 'styled-components'\nimport { easeOutQuart } from '../style/animations'\n\nconst AnimationContainer = styled.div`\n  animation-fill-mode: forwards;\n  display: flex;\n  align-items: ${({ align }) => align}}\n  justify-content: ${({ justify }) => justify}}\n  outline: none;\n  width: 100%;\n\n  &.stageIn {\n    animation: ${({ duration }) => `${duration}ms`} ${({ stageInAnimation }) =>\n  stageInAnimation} ${({ easing }) => easing};\n  }\n\n  &.stageOut {\n    animation: ${({ duration }) => `${duration}ms`} ${({ stageOutAnimation }) =>\n  stageOutAnimation} ${({ easing }) => easing};\n  }\n\n  @media (prefers-reduced-motion) {\n    &.stageIn,\n    &.stageOut {\n      animation-name: dissolve;\n    }\n  }\n`\n\nconst AnimationWrapper = ({\n  // @ts-expect-error TS(7031) FIXME: Binding element 'align' implicitly has an 'any' ty... Remove this comment to see the full error message\n  align,\n  // @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message\n  children,\n  // @ts-expect-error TS(7031) FIXME: Binding element 'dismissing' implicitly has an 'an... Remove this comment to see the full error message\n  dismissing,\n  // @ts-expect-error TS(7031) FIXME: Binding element 'duration' implicitly has an 'any'... Remove this comment to see the full error message\n  duration,\n  // @ts-expect-error TS(7031) FIXME: Binding element 'easing' implicitly has an 'any' t... Remove this comment to see the full error message\n  easing,\n  // @ts-expect-error TS(7031) FIXME: Binding element 'justify' implicitly has an 'any' ... Remove this comment to see the full error message\n  justify,\n  // @ts-expect-error TS(7031) FIXME: Binding element 'stageInAnimation' implicitly has ... Remove this comment to see the full error message\n  stageInAnimation,\n  // @ts-expect-error TS(7031) FIXME: Binding element 'stageOutAnimation' implicitly has... Remove this comment to see the full error message\n  stageOutAnimation,\n  // @ts-expect-error TS(7031) FIXME: Binding element 'onDismiss' implicitly has an 'any... Remove this comment to see the full error message\n  onDismiss,\n}) => {\n  const [content, setContent] = useState(children)\n  const [hasChanged, setHasChanged] = useState(false)\n  const [dismissed, setDismissed] = useState(false)\n  const [animationLocked, setAnimationLocked] = useState(false)\n  const [className, setClassName] = useState('stageIn')\n  const shouldUpdate = children !== content && !dismissing\n\n  function lockAnimationDuringStaging() {\n    setAnimationLocked(true)\n    setTimeout(() => {\n      setAnimationLocked(false)\n    }, duration * 2)\n  }\n\n  function stageIn() {\n    setClassName('stageIn')\n  }\n\n  function stageOut() {\n    setClassName('stageOut')\n  }\n\n  useEffect(() => {\n    if (shouldUpdate) {\n      setHasChanged(true)\n      setTimeout(() => {\n        stageIn()\n        setContent(children)\n        setHasChanged(false)\n      }, duration)\n    }\n  }, [children])\n\n  useEffect(() => {\n    if (dismissing) {\n      // we are dismissing a bit earlier then animation end to prevent accidental re-render\n      setTimeout(() => {\n        setDismissed(true)\n        onDismiss()\n      }, duration - 10)\n    }\n  }, [dismissing])\n\n  useEffect(() => {\n    lockAnimationDuringStaging()\n  }, [])\n\n  if (!animationLocked && (hasChanged || dismissing)) {\n    stageOut()\n    lockAnimationDuringStaging()\n  }\n\n  return (\n    <AnimationContainer\n      align={align}\n      duration={duration}\n      easing={easing}\n      justify={justify}\n      stageInAnimation={stageInAnimation}\n      stageOutAnimation={stageOutAnimation}\n      className={className}\n    >\n      {!dismissed && content}\n    </AnimationContainer>\n  )\n}\n\nAnimationWrapper.propTypes = {\n  align: PropTypes.string,\n  children: PropTypes.node.isRequired,\n  dismissing: PropTypes.bool,\n  duration: PropTypes.number,\n  justify: PropTypes.string,\n  easing: PropTypes.string,\n  stageInAnimation: PropTypes.shape({}).isRequired,\n  stageOutAnimation: PropTypes.shape({}).isRequired,\n  onDismiss: PropTypes.func,\n}\n\nAnimationWrapper.defaultProps = {\n  align: 'center',\n  dismissing: false,\n  duration: 300,\n  justify: 'center',\n  easing: easeOutQuart,\n  onDismiss: () => {},\n}\n\nexport default AnimationWrapper\n"]}