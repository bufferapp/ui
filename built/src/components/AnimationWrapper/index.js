export { default } from './AnimationWrapper';
export { default as useAnimation } from './hook';
//# sourceMappingURL=index.js.map