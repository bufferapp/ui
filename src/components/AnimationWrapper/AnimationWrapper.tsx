import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { easeOutQuart } from '../style/animations'

const AnimationContainer = styled.div`
  animation-fill-mode: forwards;
  display: flex;
  align-items: ${({ align }): string => align}}
  justify-content: ${({ justify }): string => justify}}
  outline: none;
  width: 100%;

  &.stageIn {
    animation: ${({ duration }): string => `${duration}ms`} ${({
  stageInAnimation,
}) => stageInAnimation} ${({ easing }): string => easing};
  }

  &.stageOut {
    animation: ${({ duration }): string => `${duration}ms`} ${({
  stageOutAnimation,
}) => stageOutAnimation} ${({ easing }): string => easing};
  }

  @media (prefers-reduced-motion) {
    &.stageIn,
    &.stageOut {
      animation-name: dissolve;
    }
  }
`

const AnimationWrapper = ({
  // @ts-expect-error TS(7031) FIXME: Binding element 'align' implicitly has an 'any' ty... Remove this comment to see the full error message
  align,
  // @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message
  children,
  // @ts-expect-error TS(7031) FIXME: Binding element 'dismissing' implicitly has an 'an... Remove this comment to see the full error message
  dismissing,
  // @ts-expect-error TS(7031) FIXME: Binding element 'duration' implicitly has an 'any'... Remove this comment to see the full error message
  duration,
  // @ts-expect-error TS(7031) FIXME: Binding element 'easing' implicitly has an 'any' t... Remove this comment to see the full error message
  easing,
  // @ts-expect-error TS(7031) FIXME: Binding element 'justify' implicitly has an 'any' ... Remove this comment to see the full error message
  justify,
  // @ts-expect-error TS(7031) FIXME: Binding element 'stageInAnimation' implicitly has ... Remove this comment to see the full error message
  stageInAnimation,
  // @ts-expect-error TS(7031) FIXME: Binding element 'stageOutAnimation' implicitly has... Remove this comment to see the full error message
  stageOutAnimation,
  // @ts-expect-error TS(7031) FIXME: Binding element 'onDismiss' implicitly has an 'any... Remove this comment to see the full error message
  onDismiss,
}) => {
  const [content, setContent] = useState(children)
  const [hasChanged, setHasChanged] = useState(false)
  const [dismissed, setDismissed] = useState(false)
  const [animationLocked, setAnimationLocked] = useState(false)
  const [className, setClassName] = useState('stageIn')
  const shouldUpdate = children !== content && !dismissing

  function lockAnimationDuringStaging() {
    setAnimationLocked(true)
    setTimeout(() => {
      setAnimationLocked(false)
    }, duration * 2)
  }

  function stageIn() {
    setClassName('stageIn')
  }

  function stageOut() {
    setClassName('stageOut')
  }

  useEffect(() => {
    if (shouldUpdate) {
      setHasChanged(true)
      setTimeout(() => {
        stageIn()
        setContent(children)
        setHasChanged(false)
      }, duration)
    }
  }, [children])

  useEffect(() => {
    if (dismissing) {
      // we are dismissing a bit earlier then animation end to prevent accidental re-render
      setTimeout(() => {
        setDismissed(true)
        onDismiss()
      }, duration - 10)
    }
  }, [dismissing])

  useEffect(() => {
    lockAnimationDuringStaging()
  }, [])

  if (!animationLocked && (hasChanged || dismissing)) {
    stageOut()
    lockAnimationDuringStaging()
  }

  return (
    <AnimationContainer
      align={align}
      duration={duration}
      easing={easing}
      justify={justify}
      stageInAnimation={stageInAnimation}
      stageOutAnimation={stageOutAnimation}
      className={className}
    >
      {!dismissed && content}
    </AnimationContainer>
  )
}

AnimationWrapper.propTypes = {
  align: PropTypes.string,
  children: PropTypes.node.isRequired,
  dismissing: PropTypes.bool,
  duration: PropTypes.number,
  justify: PropTypes.string,
  easing: PropTypes.string,
  stageInAnimation: PropTypes.shape({}).isRequired,
  stageOutAnimation: PropTypes.shape({}).isRequired,
  onDismiss: PropTypes.func,
}

AnimationWrapper.defaultProps = {
  align: 'center',
  dismissing: false,
  duration: 300,
  justify: 'center',
  easing: easeOutQuart,
  onDismiss: () => {},
}

export default AnimationWrapper
