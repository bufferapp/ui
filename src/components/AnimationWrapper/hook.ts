import { useState } from 'react'
import AnimationWrapper from './AnimationWrapper'

// @ts-expect-error TS(7006) FIXME: Parameter 'props' implicitly has an 'any' type.
function useAnimation(props) {
  const [dismissing, setDismissing] = useState(false)
  const animationProps = { ...props, dismissing }

  return {
    AnimationWrapper,
    animationProps,
    dismiss: () => setDismissing(true),
  }
}

export default useAnimation
