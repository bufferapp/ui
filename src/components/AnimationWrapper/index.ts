export { default } from './AnimationWrapper'
export useAnimation from './hook'
