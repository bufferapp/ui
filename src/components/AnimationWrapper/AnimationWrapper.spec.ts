// @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'jest... Remove this comment to see the full error message
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import AnimationWrapper from './AnimationWrapper'

snap(AnimationWrapper, './AnimationWrapper.jsx')
