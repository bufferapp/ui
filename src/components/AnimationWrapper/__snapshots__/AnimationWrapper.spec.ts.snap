// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > AnimationWrapper Matches snapshot when boolean prop "dismissing" is set to: "false" 1`] = `
<styled.div
  align="jest-auto-snapshots String Fixture"
  className="stageIn"
  duration={1}
  easing="jest-auto-snapshots String Fixture"
  justify="jest-auto-snapshots String Fixture"
  stageInAnimation={Object {}}
  stageOutAnimation={Object {}}
>
  <NodeFixture />
</styled.div>
`;

exports[`jest-auto-snapshots > AnimationWrapper Matches snapshot when passed all props 1`] = `
<styled.div
  align="jest-auto-snapshots String Fixture"
  className="stageOut"
  duration={1}
  easing="jest-auto-snapshots String Fixture"
  justify="jest-auto-snapshots String Fixture"
  stageInAnimation={Object {}}
  stageOutAnimation={Object {}}
>
  <NodeFixture />
</styled.div>
`;

exports[`jest-auto-snapshots > AnimationWrapper Matches snapshot when passed only required props 1`] = `
<styled.div
  align="center"
  className="stageIn"
  duration={300}
  easing="cubic-bezier(0.25, 1, 0.5, 1)"
  justify="center"
  stageInAnimation={Object {}}
  stageOutAnimation={Object {}}
>
  <NodeFixture />
</styled.div>
`;
